### PR TITLE
Update use of cml cli

### PR DESCRIPTION
**Pull request by tasdomas**
The cli for cml has been updated, it's best if our examples use the new command structure.

### CODE REVIEW

File: example-get-started/code/.github/workflows/cml.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ jobs:
       - uses: iterative/setup-dvc@v1
       - uses: actions/checkout@v3
         with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (shcheklein) [inline example-get-started/code/.github/workflows/cml.yaml:11]: @tasdomas Q: why do we need this?

Comment (tasdomas) [reply example-get-started/code/.github/workflows/cml.yaml:11]: For `cml comment create --pr` (comments on PRs) to work, cml needs to be able to determine the PR id based on the HEAD commit. However github runs workflows on a {master..PR} merge commit, which cml currently cannot attribute to the actual PR. This is a temporary workaround until we implement proper PR lookup in cml.

Comment (shcheklein) [reply example-get-started/code/.github/workflows/cml.yaml:11]: got it. In this repo we were also publishing comments to the main branch commits:

https://github.com/iterative/example-repos-dev/pull/140/files#diff-74446cb71a98e9596cef50d831916c75ed4f0c65a5083c75dca8de15cdd5dd50R18-R22

what will happen to those with this new change (and with `--pr`)?

I mean reports like this https://github.com/iterative/example-get-started/commit/a470fae5c0ccf71641e10d99e69e97534c3d811e#commitcomment-84532489

Comment (tasdomas) [reply example-get-started/code/.github/workflows/cml.yaml:11]: There's an issue with commit PRs, that affects CML - in github they [no longer appear in the PR timeline](https://github.blog/changelog/2022-08-04-commit-comments-no-longer-appear-in-the-pull-request-timeline/) which may make adding commit comments in our example repos fall short in showcasing CML functionality.

@casperdcl  - what do you think?

Comment (casperdcl) [reply example-get-started/code/.github/workflows/cml.yaml:11]: > In this repo we were also publishing comments to the main branch commits

afaik `--pr` should fallback to non-`--pr` in such cases.

Alternatively should we change this workflow from `on: push` to `on: pull_request` (or both `push` & `pull_request`)? To me the `main` monitoring use case is different from the PR use case, so maybe even separate workflows?

Comment (shcheklein) [reply example-get-started/code/.github/workflows/cml.yaml:11]: yep, would be a bit annoying to duplicate the workflow file in this repo in this simple case though

Comment (tasdomas) [reply example-get-started/code/.github/workflows/cml.yaml:11]: I've revisited this and if all we want is commit tracking, then we don't need this change. Sorry, I was a tad over-zealous.

File: example-get-started/code/.github/workflows/cml.yaml [hunk 2]
```diff
@@ -20,20 +20,24 @@ jobs:
...
+          cat <<EOF > report.md
+          # CML Report
+          ## Plots
+          ![Precision & Recall](./prc.svg)
+          ![Confusion Matrix](./confusion.svg)
+
+          ## Metrics and Params
+          ### ${PREVIOUS_REF} → workspace
+          EOF
+
```
Comment (casperdcl) [inline example-get-started/code/.github/workflows/cml.yaml:32]:
```suggestion
```

File: example-get-started/code/.github/workflows/cml.yaml [hunk 3]
```diff
@@ -20,20 +20,24 @@ jobs:
...
-          cml publish prc.svg --title "Precision & Recall" --md >> report.md
 
           dvc plots diff $PREVIOUS_REF workspace \
             --show-vega --targets evaluation/plots/confusion_matrix.json > vega.json
           vl2svg vega.json confusion.svg
-          cml publish confusion.svg --title "Confusion Matrix" --md >> report.md
 
-          echo "## Metrics and Params" >> report.md
-          echo "### $PREVIOUS_REF → workspace" >> report.md
           dvc exp diff $PREVIOUS_REF --show-md >> report.md
```
Comment (casperdcl) [inline example-get-started/code/.github/workflows/cml.yaml:41]:
```suggestion
          cat <<EOF > report.md
          # CML Report
          ## Plots
          ![Precision & Recall](./prc.svg)
          ![Confusion Matrix](./confusion.svg)

          ## Metrics and Params
          ### ${PREVIOUS_REF} → workspace
          $(dvc exp diff $PREVIOUS_REF --show-md)
          EOF
```

**Review by casperdcl — APPROVED** [review]
minor suggestion to tidy code a bit

**Comment by shcheklein** [comment]
@iterative/cml what is the status of this?

**Comment by tasdomas** [comment]
@shcheklein - once this is landed, how do we release the changes? I suspect I will not have access to all the necessary cloud resources

File: example-get-started/code/.github/workflows/cml.yaml (hunk 3 above)
Comment (tasdomas) [reply example-get-started/code/.github/workflows/cml.yaml]: executing a command in the middle of a heredoc gives me the shivers

Comment (tasdomas) [reply example-get-started/code/.github/workflows/cml.yaml]: Submitted an alternative - ptal

**Comment by tasdomas** [comment]
@shcheklein the functionality will remain unchanged - the comments will be created on pushed commits

**Comment by shcheklein** [comment]
thanks @tasdomas @casperdcl ... I'll update the repo (I'm stuck a bit since there is something happening with DVC)

**Pull request merged**